eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -239,6 +241,14 @@ def to_side(self):
     REGEX_CHAR: /[^\\^$().*+?{}\[\]|]/
 """
 
+OBSERVABLES_GRAMMAR = """
+    observable: CNAME ":" pattern_mod? (basic_observable | scaled_observable) pattern_quantified?
+    basic_observable: const? (value | abstract_sequence) (DOUBLE_COLON compartment)?
+    !scaled_observable: const | basic_observable | scaled_observable "+" scaled_observable | scaled_observable "-" scaled_observable | scaled_observable "*" scaled_observable | scaled_observable "/" scaled_observable | scaled_observable POW const | "(" scaled_observable ")"
+    !pattern_quantified: (">" | "<" | ">=" | "<=" | "==" | "!=") INT
```

Comment:
Done in commit [6da78d9](https://github.com/sybila/eBCSgen/pull/109/commits/6da78d948ae0d57a525081c424f50b25b652295c). Decided to use `complex` instead of `rate_complex` because I mistook `rate_complex` for `rate_agent` and thought it had square brackets `[]`. 

`rate_complex` is a subset of `complex`: 
```
complex: (abstract_sequence|value|cmplx_name) DOUBLE_COLON compartment
rate_complex: value DOUBLE_COLON compartment
```

Is it ok or should I change it to `rate_complex`?

Reply:
You are right, `complex` is even better, it allows to use aliases and the nesting.